AdvLoggerPkg/Application/DecodeUefiLog/DecodeUefiLog.py

```diff
@@ -598,22 +595,123 @@ def _GetLines(self, lines, LoggerInfo):
         CurrentLine = LoggerInfo["CurrentLine"]
         StartLine = LoggerInfo["StartLine"]
 
-        while (Status == self.SUCCESS):
-            (Status, MessageLine) = self._GetNextFormattedLine(MessageLine, LoggerInfo)
-            if Status != self.SUCCESS:
-                if Status != self.END_OF_FILE:
-                    print(f"Error {Status} from GetNextFormattedLine")
-                break
+        if LoggerInfo["Version"] == self.V4_LOGGER_INFO_VERSION:
+            # There is a potential that we may have enabled auto wrap.

```

Comment:
I think the main suggestion is there is a lot going on here that makes it hard to understand what is going on. I would break all of this into a few distinct functions and then provide documentation for each function. This logic really does three things. 1. Find the first uncorrupted block, 2. Generate the list of messages, 3. Turn messages into strings

1. Turn the logic that finds the offset of the first uncorrupted block of records into it's own function.

`StartOffset = FirstValidBlock(LoggerInfo)`

2. Turn the logic that builds the list of Message Entries into it's own function.

`MessageEntries = ParseMessageEntries(LoggerInfo, StartOffset)`

3. Convert the Message entries into a list of lines

`Lines = BuildLinesFromMessages(MessageEntries)`

It does not have to be exactly this, but by breaking it up, and then documenting each function, I think it will make it easier for someone new to understand what is going on.

Reply:
Updated to closer to your recommendation. Thanks for the suggestion!